src/html_support_files/odoc_search.js

```diff
@@ -0,0 +1,34 @@
+function createWebWorker() {
+  var searchs = search_urls.map((search_url) => {
+    let parts = document.location.href.split("/");
+    parts[parts.length - 1] = search_url;
+    return parts.join("/");
+  });
+  blobContents = ['importScripts("' + searchs.join(",") + '");'];
+  var blob = new Blob(blobContents, { type: "application/javascript" });
+  var blobUrl = URL.createObjectURL(blob);
+
+  var worker = new Worker(blobUrl);
+  URL.revokeObjectURL(blobUrl);
+
+  return worker;
+}
```

Comment:
Welcome to the ridiculous world of web programming. Been there. You should add a comment that you are making that dance to enable webworking over the `file://` protocol.